crates/cairo-profiler/src/trace_reader/function_trace_builder/function_stack_trace.rs

```diff
@@ -1,37 +1,37 @@
+use crate::trace_reader::function_name::FunctionName;
 use crate::trace_reader::function_trace_builder::Steps;
-use crate::trace_reader::functions::FunctionName;
 
-pub(super) struct Function {
+struct FunctionOnStack {
     pub name: FunctionName,
     /// Steps of the function at the moment of putting it on the stack.
     pub steps: Steps,
     /// Consecutive recursive calls to this function that are currently on the stack.
     recursive_calls_count: usize,
 }
 
-/// The function stack trace of the current function, excluding the current function.
-pub(super) struct FunctionStack {
-    stack: Vec<Function>,
-    /// Tracks the depth of the function stack, without limit. This is usually equal to
+/// The function call stack of the current function, excluding the current function call.
+pub(super) struct CallStack {
+    stack: Vec<FunctionOnStack>,
+    /// Tracks the depth of the function call stack, without limit. This is usually equal to
     /// `stack.len()`, but if the actual stack is deeper than `max_function_trace_depth`,
     /// this remains reliable while `stack` does not.
     real_function_stack_depth: usize,
     /// Constant through existence of the object.
-    max_function_trace_depth: usize,
+    max_function_stack_trace_depth: usize,
 }
 
-pub(super) enum FunctionType {
-    Regular(Function),
+pub(super) enum CallType {
```

Comment:
Can me move it above the function that returns it?

Can we document the elements?